### DISCUSSION TITLE
Use curl to send a DoH request

### DISCUSSION THREAD

**Discussion by Heshmatkhah** [AdguardTeam/AdGuardHome#7475] [Q&A] [answered]
Hi
I need to send DoH requests in a script on an embedded device that I can't install any package or change DNS configuration,

According to [Cloudflare](https://developers.cloudflare.com/1.1.1.1/encryption/dns-over-https/make-api-requests/dns-json/#examples) I can use the following command to send a DoH request:
```shell
curl --header "accept: application/dns-json" "https://cloudflare-dns.com/dns-query?name=example.com&type=AAAA" | jq
```
![image](https://github.com/user-attachments/assets/34a80ea7-b808-40d6-ba25-2af5c2a226ac)

but the same command doesn't work for AdGuard Home:
```shell
curl -k --header "accept: application/dns-json" "https://dns.myhome.com/dns-query?name=example.com&type=AAAA"

> Bad Request
```

This is while the DoH is working correctly and the curl can resolve names using it:
```shell
curl --doh-url "https://dns.mazhome.ir/dns-query" https://example.com
```
After above command this added to my logs (-> AdGuard Home is up and working!):
![image](https://github.com/user-attachments/assets/9105df23-948c-4acf-bf82-b35a40e54525)

So, how to use curl to get **JSON** DoH response from AdGuard Home?

**Comment by schzhn** [comment]
```
curl -v example.com --doh-url https://127.0.0.1/dns-query &> /dev/stdout | grep "* DoH"

dig +https @127.0.0.1 example.com
```

**Comment by Heshmatkhah** [comment]
Hi
Thank you for your response
I don't have dig on that platform, and i can't install it
Is there any other way than using grep on curl logs?
does the server support json based DoH?

**Reply by schzhn** [reply, marked as answer]
At the current moment, no
https://github.com/AdguardTeam/dnsproxy/issues/422
